src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs

```diff
@@ -118,7 +120,24 @@ public SdkResult GetSdkResult(int submissionId, SdkReference sdk, LoggingContext
                     SdkResultFactory resultFactory = new SdkResultFactory(sdk);
                     try
                     {
-                        SdkResult result = (SdkResult) sdkResolver.Resolve(sdk, context, resultFactory);
+                        SdkResult result;
+
+                        try
+                        {
+                            result = (SdkResult) sdkResolver.Resolve(sdk, context, resultFactory);
+                        }
+                        catch (Exception e) when (sdkResolver is NuGetSdkResolver && e is FileNotFoundException)
+                        {
+                            // MSB4243: The NuGet-based SDK resolver failed to run because NuGet assemblies could not be located.  Check your installation of MSBuild or set the environment variable "{0}" to the folder that contains the required NuGet assemblies. {1}
+                            loggingContext.LogWarning(null, new BuildEventFileInfo(sdkReferenceLocation), "CouldNotRunNuGetSdkResolver", NuGetSdkResolverBase.NuGetAssemblyPathEnvironmentVariableName, e.Message);
+                            continue;
+                        }
+                        catch(Exception e)
+                        {
+                            // MSB4242: The SDK resolver "{0}" failed to run. {1}
+                            loggingContext.LogWarning(null, new BuildEventFileInfo(sdkReferenceLocation), "CouldNotRunSdkResolver", sdkResolver.Name, e.Message);
```

Comment:
What were we doing for this before? Is it ok that this is no longer fatal?

Reply:
Yes before it was a fatal error.  When testing things out, a fatal build error when something was wrong with a resolver seemed like overkill.  What do you think?